contribute/design/1.x/cstor/pool/pool_expansion_by_expanding_disk.md

```diff
@@ -0,0 +1,135 @@
+
+## Motivation
+
+There were cases where OpenEBS users expand the pool by expanding underlying disks(which were used by the pool) when space on the pool got exhausted.
+
+### Goals
+
+- Expand cStor pool when the underlying disk got expanded.
+
+### Non-Goals
+
+- If the user dedicated the first part of the partitioned disk for cStor and second part for third-party then pool expansion can't be performed automatically until and unless the user extends the first partition.
+
+## Proposal
+
+### User Stories
+
+As an OpenEBS user, I should be able to expand the pool when the underlying disk got expanded.
+
+### Current Implementation
+
+- When the user creates the CSPC, the CSPC controller which is watching for CSPC will get create requests and creates cstorpoolinstances and cstorpooldeployments. CstorPoolInstance is responsible for holding the blockdevice information and other pool configurations whereas cStor pool deployment have 3 containers cstor-pool-mgmt, cstor-pool and m-exporter as the name conveys cstor-pool container creates pool(ZFS file system) on top of the blockdevices, whereas cstor-pool-mgmt sidecar container is responsible for managing these pool. If any changes required as of day 2 operations on the pool then pool configurations will be updated by adding BlockDevices, adding RaidGroup and replacing blockdevices cstor-pool-mgmt looks for any pending pool operations for CStorPoolInstance then performs pending operation on the pool.
+
+Below is high-level info to explain the Day2 operations that are currently supported.
+
+- Pool Expansion When BlockDevice/raid group added:
+	cstor-pool-mgmt will iterate over all the BlockDevices that exist in CStorPoolInstance resource and verify whether the corresponding BlockDevice/raid group is part of the pool. If that BlockDevice/raid group is not part of the pool then pool expansion will be triggered on newly added BlockDevice/raid group.
+
+- BlockDevice Replacement In Pool:
+    cstor-pool manager will iterate over all the BlockDevices present in CstorPoolInstance and verifies whether the current blockdevice is for BlockDevice replacement request[How cstor-pool-mgmt can know? We are maintaining annotation called openebs.io/predecessor: <replaced_blockdevice_name> on the claim of that blockdevice]? If requested BlockDevice is valid for replacement then the pool manager will perform BlockDevice replacement operation on the pool.
+
+### Proposed Implementations
+
+- CStor-pool Manger inbuilt CSPI controller watches for CSPI resources and tries to get the desired state via every sync operation. As part of reconciliation logic, it will iterate over all the BlockDevices that exist on CStorPoolInstance resource if it found BlockDevice resource capacity(considerable changes) is greater than the existing capacity of BlockDevice on CstorPoolInstanceBlockDevice then we will follow below-mentioned steps to perform the [pool expansion](#Pool_Expansion_Steps).
```

Comment:
1. typo `Manager`
2. What is meant by considerable change. Are we setting some limits like expand pool only if size increase > 10GB
3. `CstorPoolInstanceBlockDevice` Does CSPI keep a copy of the BlockDevice resource?

Reply:
> What is meant by considerable change. Are we setting some limits like expand pool only if size increase > 10GB

Here, a considerable change in the sense of more than 500MB(I will update the doc).

> CstorPoolInstanceBlockDevice Does CSPI keep a copy of the BlockDevice resource?

No, It will following information 
```go
// CStorPoolInstanceBlockDevice contains the details of block devices that
// constitutes a raid group.
type CStorPoolInstanceBlockDevice struct {
	// BlockDeviceName is the name of the block device.
	BlockDeviceName string `json:"blockDeviceName"`
	// Capacity is the capacity of the block device.
	// It is system generated
	Capacity string `json:"capacity"`
	// DevLink is the dev link for block devices
	DevLink string `json:"devLink"`
}
```